From ac524bd95674207cb2626ffd9ef1e4eb6ecee85e Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Tue, 29 Apr 2025 10:44:19 +0100
Subject: [PATCH 1/2] Search for Catch2 with find_package() If not found,
 download it with the FetchContent module.

This changes were based on the following commit:
https://github.com/TartanLlama/expected/pull/147/commits/c94d06e3fcffd79e09753aa07cdaf60f37a45d49

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddab64e..e2c897f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,9 +67,13 @@ if(EXPECTED_BUILD_TESTS)
   set(CATCH_INSTALL_HELPERS OFF)
   set(CATCH_BUILD_TESTING OFF)
   set(CATCH_INSTALL_DOCS OFF)
-  FetchContent_Declare(Catch2 URL
-    https://github.com/catchorg/Catch2/archive/v2.13.10.zip) 
-  FetchContent_MakeAvailable(Catch2)
+
+  find_package(Catch2)
+  if (NOT Catch2_FOUND)
+    FetchContent_Declare(Catch2 URL
+      https://github.com/catchorg/Catch2/archive/v2.13.10.zip)
+    FetchContent_MakeAvailable(Catch2)
+  endif()
 
   file(GLOB test-sources CONFIGURE_DEPENDS tests/*.cpp)
   list(FILTER test-sources EXCLUDE REGEX "tests/test.cpp")
-- 
2.34.1

